src/main/java/io/anserini/util/DumpIndex.java

```diff
@@ -0,0 +1,221 @@
+      Document d = reader.document(i);
+      IndexableField id = d.getField(IndexWebCollection.FIELD_ID);
+      String idStr = id.stringValue();
+      if (mapping.containsKey(idStr)) {
+        mapping.put(idStr, i);
+        total--;
+        if (total == 0) break;
+      }
+    }
+    for (String s : externalIDs) {
+      sb.append(mapping.get(s)+"\n");
+    }
+    return sb.toString();
+  }
+
+  String print_document_name( DirectoryReader reader, int number ) throws IOException {
+    StringBuilder sb = new StringBuilder();
+    Document d = reader.document(number);
+    IndexableField id = d.getField(IndexWebCollection.FIELD_ID);
+    sb.append(id.stringValue())
+      .append("\n");
+    return sb.toString();
+  }
+
+  String print_document_text( DirectoryReader reader, int number ) throws IOException, RawDocNotStoredException {
+    StringBuilder sb = new StringBuilder();
+    Document d = reader.document(number);
+    IndexableField id = d.getField(IndexWebCollection.FIELD_BODY);
+    if (id == null) {
+      throw new RawDocNotStoredException("Raw Contents not Stored!");
+    }
+    sb.append(id.stringValue())
+      .append("\n");
+    return sb.toString();
+  }
+
+  String print_document_vector( DirectoryReader reader, int number ) throws IOException, DocVectorNotStoredException {
+    StringBuilder sb = new StringBuilder();
+    Terms terms = reader.getTermVector(number, IndexWebCollection.FIELD_BODY);
+    if (terms == null) {
+      throw new DocVectorNotStoredException("Doc Vector not Stored!");
+    }
+    TermsEnum te = terms.iterator();
+    if (te == null) {
+      throw new DocVectorNotStoredException("Doc Vector not Stored!");
+    }
+    while((te.next()) != null){
+      sb.append(te.term().utf8ToString()+" "+te.totalTermFreq()+"\n");
+    }
+    return sb.toString();
+  }
+
+  void argsEnough(int required, int n) {
+    if (n < required) {
+      printUsage();
+      System.exit(1);
+    }
+  }
+
+  void printUsage() {
+    StringBuilder sb = new StringBuilder();
+    sb.append("DumpIndex <repository> <command> [ <argument> ]*\n")
+      .append("These commands retrieve data from the repository: \n")
+      .append("    Command              Argument       Description\n")
+      .append("    term (t)             Term text      Print inverted list for a term\n")
+      .append("    documentid (di)      Original IDs   Print the internal document IDs of documents\n")
+      .append("    documentname (dn)    Internal Document ID    Print the text representation of a document ID\n")
+      .append("    documenttext (dt)    Internal Document ID    Print the text of a document\n")
+      .append("    documentvector (dv)  Internal Document ID    Print the document vector of a document\n")
+      .append("    stats (s)                           Print statistics for the Repository\n");
+    System.out.println(sb.toString());
+  }
+
+  public static void main(String[] clArgs) throws IOException{
```

Comment:
At the beginning I tried to not to use args4j to make the code with less dependencies.
But I can migrate to args4j to make it more compatible with other codes in the repo.